packages/contract-utils/role-transfer/src/storage.rs

```diff
@@ -0,0 +1,78 @@
+use soroban_sdk::{panic_with_error, Address, Env, IntoVal, Val};
+
+use crate::RoleTransferError;
+
+/// Initiates the role transfer (owner/admin).
+/// If `live_until_ledger == 0`, cancels the pending transfer.
+///
+/// Does not emit any events.
+///
+/// Returns None if the transfer was initiated, Some(Address) where the Address
+/// is the new candidate for the pending role in the storage if the pending
+/// transfer is revoked.
+pub fn transfer_role<T, U>(
+    e: &Env,
+    current: &Address,
+    new: &Address,
+    active_key: &T,
+    pending_key: &U,
+    live_until_ledger: u32,
+) -> Option<Address>
+where
+    T: IntoVal<Env, Val>,
+    U: IntoVal<Env, Val>,
+{
+    current.require_auth();
+
+    let current_in_storage = e
+        .storage()
+        .instance()
+        .get::<_, Address>(active_key)
```

Comment:
```suggestion
        .get::<T, Address>(active_key)
```
so that we use it explicitly :) feel free to ignore it